### ISSUE TITLE
relative Options.BaseURI breaks in-document $ref resolution

### ISSUE THREAD

**Issue by tdakkota** [ogen-go/schemacompiler#28]
A relative `Options.BaseURI` (e.g. `schema.yml`) makes every in-document `$ref` dangle. Registration keys on the raw base URI while lookup uses the RFC 3986-resolved one, so the two never meet.

Reproduced on `main` (9779cbb) with this schema:

```json
{
  "type": "object",
  "properties": {"a": {"$ref": "#/$defs/A"}},
  "$defs": {"A": {"type": "string"}}
}
```

| `Options.BaseURI` | result |
| --- | --- |
| `""` | clean |
| `"schema.yml"` | `SeverityError` |
| `"/schema.yml"` | clean |
| `"https://ex.com/schema.yml"` | clean |

The failing diagnostic:

```
/properties/a: unresolved $ref #/$defs/A: pointer "/$defs/A" not found in resource "/schema.yml"
```

Note the resource is reported as `/schema.yml` — resolved — while registration used `schema.yml` verbatim. Relevant sites: `internal/frontend/convert.go:551` (`register`, keyed on `nearestBaseURI`) and `internal/frontend/resolve.go:93,123` (lookup over `st.refBaseURI`).

`cmd/schemac -base-uri` passes the flag through verbatim, so a user pointing it at a bare filename hits this directly.

Fix is presumably to normalize the base URI once at entry (`schemacompiler.go:61`) so registration and lookup agree, rather than resolving on only one side.

Found incidentally while working on #21; not touched there.
